src/tyr/serializers.cc

```diff
@@ -179,6 +179,34 @@ waypoint(const valhalla::Location& location, bool is_tracepoint, bool is_optimiz
   return waypoint;
 }
 
+/*
+ * This function serializes the via_waypoints object on the leg.
+ */
+valhalla::baldr::json::MapPtr via_waypoint(const valhalla::Location& location,
```

Comment:
Would it be possible to add some unittests of this and the `via_waypoints` below in this file to exercise them some more?